code/ui/components/slider/Slider.cs

```diff
@@ -0,0 +1,182 @@
+using System;
+using System.Linq;
+
+using Sandbox;
+using Sandbox.UI;
+using Sandbox.UI.Construct;
+
+namespace TTTReborn.UI
+{
+    /// <summary>
+    /// A horizontal slider. Can be float or whole number.
+    /// </summary>
+    public class Slider : TTTPanel
+    {
+        public Panel Track { get; protected set; }
+        public Panel TrackInner { get; protected set; }
+        public Panel Thumb { get; protected set; }
```

Comment:
Should these be public? What's the use case for accessing them outside. Changing the style?

Reply:
Retrieving data off of them.